cli/tools/lint.rs

```diff
@@ -69,11 +110,29 @@ pub async fn lint_files(
   };
   let reporter_lock = Arc::new(Mutex::new(create_reporter(reporter_kind)));
 
+  // Try to get configured rules. CLI flags take precendence
+  // over config file, ie. if there's `rules.include` in config file
+  // and `--rules-include` CLI flag, only the flag value is taken into account.
+  // TODO(bartlomieju): this is done multiple times for each file because
+  // Vec<Box<dyn LintRule>> is not clonable, this should be optimized.
+  get_configured_rules(
+    maybe_lint_config.as_ref(),
+    rules_tags.clone(),
+    rules_include.clone(),
+    rules_exclude.clone(),
+  )?;
```

Comment:
what is this call for?

Reply:
This is validating that provided configuration actually resolves to at least 1 rule - otherwise it will error out saying that there are no configured rules. This is a code smell, but it's the only place this can be done, before parallelizing work across threads. It's done again for each file later, but at that point it's infallible.

Reply:
ah that makes sense!